### PR TITLE
SOLR-15949: Docker, use Java 17

### PR DESCRIPTION
**Pull request by dsmiley**
azul/zulu-openjdk-alpine:17-jre-headless

Also switched OS from Debian to Alpine.  WIP

https://issues.apache.org/jira/browseSOLR-15949

It wasn't easy to find a Java 17 *JRE* in Docker hub but I ultimately did with Azul.  I trust them.

In my attempt to do this, I wound up choosing Alpine, which is another, perhaps bigger change.  I'd be happy to move this off to another issue; I really should but I'm pushing this now to show what I have since I put time into it.  I think most edits and possible further TODO in this PR are due to Alpine.
* I added `su-exec` instead of `gosu` which I found easier to install, and I hear it's smaller.  I removed to gosu test; honestly I'm not sure where it's used -- if nowhere then maybe we don't need a test.
* init-var-solr was using `cp -a ...` but I saw warnings because the owner couldn't be preserved, which makes sense.  This could be done independently of this PR; it may not be related but it was a warning I saw.
* I tested `gradlew testDocker` and it passes.
* I suspect the official image building might not work because I didn't find all of the unix add-ons when converting to `apk`.  I didn't try it yet; this is a TODO.

### CODE REVIEW

**Comment by dsmiley** [comment]
I saw this JIRA issue about `lsof` https://issues.apache.org/jira/browse/SOLR-13089 that was closed because we moved away from Alpine once but may be relevant again.

**Comment by dsmiley** [comment]
FYI https://hub.docker.com/r/azul/zulu-openjdk-alpine/tags
JRE vs JDK saves almost 80MB which is a ton.  Interestingly the "-headless" seems to change nothing for the image sizes.  Note that the numbers there are compressed numbers.  When it gets to your machine, these expand significantly.

File: solr/docker/build.gradle [hunk 1]
```diff
@@ -27,7 +27,7 @@ apply plugin: 'base'
 def dockerImageRepo = propertyOrEnvOrDefault("solr.docker.imageRepo", "SOLR_DOCKER_IMAGE_REPO", "apache/solr")
 def dockerImageTag = propertyOrEnvOrDefault("solr.docker.imageTag", "SOLR_DOCKER_IMAGE_TAG", "${version}")
 def dockerImageName = propertyOrEnvOrDefault("solr.docker.imageName", "SOLR_DOCKER_IMAGE_NAME", "${dockerImageRepo}:${dockerImageTag}")
-def baseDockerImage = propertyOrEnvOrDefault("solr.docker.baseImage", "SOLR_DOCKER_BASE_IMAGE", 'openjdk:11-jre-slim')
+def baseDockerImage = propertyOrEnvOrDefault("solr.docker.baseImage", "SOLR_DOCKER_BASE_IMAGE", 'openjdk:17-slim')
```
Comment (janhoy) [inline solr/docker/build.gradle:30]:
```suggestion
def baseDockerImage = propertyOrEnvOrDefault("solr.docker.baseImage", "SOLR_DOCKER_BASE_IMAGE", 'eclipse-temurin:17-jre')
```

**Review by janhoy — COMMENTED** [review]
Propose we use `eclipse-temurin:17-jre` which is an official jre image, slimmer than `openjdk:17-slim`.

**Review by janhoy — APPROVED** [review]

**Comment by janhoy** [comment]
Tested `SOLR_DOCKER_BASE_IMAGE=eclipse-temurin:17-jre ./gradlew dockerBuild` and started the resulting image. Worked great. Just add CHANGES and possibly major-changes entries and we shuold be good.

**Pull request merged**